addon/components/ember-thead/component.js

```diff
@@ -225,17 +226,22 @@ export default Component.extend({
     this.columnTree.set('enableReorder', this.get('enableReorder'));
   },
 
+  _onColumnsChange() {
+    if (this.get('columns.length') === 0) {
```

Comment:
When would a table have 0 columns?

Reply:
I'm relatively certain this was to address issues with the testing scenario whcih has a button that removes a column, if you continue to remove columns after you reach 0 you get a unrelated error.

Reply:
In our app usage, sometimes the data provided to an `{{ember-table}}` is empty for a little while, for instance when the source data for a table is refreshed. Without this check, ET2 will throw an exception in those cases where there are (temporarily) 0 columns

Reply:
Reviewed in-person with @mixonic . We hypothesized that the issues we see in our app may be due to interim state, where a `columns` property is set to `[]` for a moment (before later on being set to the non-empty server-provided data) and the observers in `ember-thead` fire eagerly. If that were the case, the `_onColumnsChange` hook here might be firing multiple times before the columns data has "settled" to its final state in its runloop.
I tried using a counter variable in this method that increments on each call and decrements in the `schedule`d callback, only doing work when it decrements back to 0, but that didn't solve the issues. It may be a mistake in our app that causes this, but we certainly run into a state where `columns.length` is 0 when we call `fillupHandler`, which causes ember-table to throw (see below).

After a bit of research, it looks like the `ColumnTreeNode` code (in `column-tree`) makes the incorrect assumption that any column that `isLeaf` cannot be the root, and any non-`isLeaf` node must have subcolumnNodes under it. When we have 0 columns, though, we end up with only a single `ColumnTreeNode` that is both the root *and* `isLeaf`. The `ensureWidthConstraint` method there runs into trouble because its assumptions about leaf vs non-leaf nodes breaks down in this case. This seems like an issue that should be fixed, but the fix can be separated from this particular PR. This PR improves the situation for adding/removing columns and doesn't fix the issue with 0 columns (but doesn't make it any worse, either).